### PR TITLE
Rename methods according to the lc.noun and lc.verb() rule

### PR DESCRIPTION
**Pull request by barentsen**
Over in #286 the idea was raised to give our properties and methods more consistent names, by ensuring that properties use nouns (e.g. `lc.ra`) and that methods use verbs (e.g. `lc.flatten()`).

This PR shows the most significant changes that would be needed to implement this change.  Tutorials will need to be updated as well.

### CODE REVIEW

File: lightkurve/convenience.py [hunk 1]
```diff
@@ -9,22 +9,22 @@
 
 
 def cdpp(flux, **kwargs):
```
Comment (christinahedges) [inline lightkurve/convenience.py]: Should we have something that is a method (verb) also be allowed to be a property (noun)? I think this is more confusing? It's estimating every time.

File: lightkurve/targetpixelfile.py [hunk 2]
```diff
@@ -833,7 +826,7 @@ def prf_photometry(self, cadences=None, parallel=True, **kwargs):
         return prfphot
 
     def prf_lightcurve(self, **kwargs):
```
Comment (christinahedges) [inline lightkurve/targetpixelfile.py]: What is a `prf_lightcurve`? It feels like we only need `extract_prf_photometry`? There's no equivalent `sap_lightcurve`.

File: lightkurve/convenience.py (hunk 1 above)
Comment (barentsen) [reply lightkurve/convenience.py]: Agreed!  Will change.

File: lightkurve/targetpixelfile.py (hunk 2 above)
Comment (barentsen) [reply lightkurve/targetpixelfile.py]: I am not going to rename the PRF methods for now and move them to a separate package instead.

**Comment by barentsen** [comment]
I have now also updated the tutorial notebooks where necessary.  This should be good to merge if we agree on the new names!

**Comment by barentsen** [comment]
@gully That's a very valid concern.  I feel like we can still do this now because we are only 6 months in and still in beta, and because we aren't changing names that are very popular.  Of the changes made, I expect that only the renaming of `centroids()` and `cdpp()` would be noticed by users.  We could keep those names and add a `DeprecationWarning` if we think it is important to do so?

**Comment by mirca** [comment]
@gully That concern makes sense to me. But I think as long as we deliver a very nice release notes for the next version the net gain of having a more consistent API will be worth it.

**Pull request merged**